test/cpp/interop/interop_server.cc

```diff
@@ -136,6 +139,26 @@ bool CheckExpectedCompression(const ServerContext& context,
   return true;
 }
 
+void RecordMetrics(grpc::experimental::CallMetricRecorder* recorder,
+                   const xds::data::orca::v3::OrcaLoadReport& request_metrics) {
+  // Do not record when zero since it indicates no test per-call report.
+  if (request_metrics.cpu_utilization() > 0) {
+    recorder->RecordCpuUtilizationMetric(request_metrics.cpu_utilization());
+  }
+  if (request_metrics.mem_utilization() > 0) {
+    recorder->RecordMemoryUtilizationMetric(request_metrics.mem_utilization());
+  }
+  if (request_metrics.rps_fractional() > 0) {
+    recorder->RecordQpsMetric(request_metrics.rps_fractional());
+  }
+  for (const auto& p : request_metrics.request_cost()) {
+    recorder->RecordRequestCostMetric(p.first.c_str(), p.second);
```

Comment:
Are you sure this works correctly?  This is storing `absl::string_view` keys in the `CallMetricRecorder` that point to storage in the request proto, but I don't think that request proto is guaranteed to live until the ORCA metadata is serialized to be sent back to the client, so it seems like this could leave pointers dangling into unsafe memory.

In our e2e tests, we deal with this by arena-allocating a copy of these strings:

https://github.com/grpc/grpc/blob/c5213ceff9edc723cbf56a5f917083ff4a025bf0/test/cpp/end2end/client_lb_end2end_test.cc#L141

Reply:
Copied. Was not familiar with this API.